lib/cylc/network/client.py

```diff
@@ -270,16 +295,26 @@ def _timeout_handler(suite, host, port):
             raise ClientError('Suite "%s" already stopped' % suite)
 
     @classmethod
-    def get_location(cls, suite, owner, host):
-        """Extract host and port from a suite's contact file."""
+    def get_location(cls, suite: str, owner: str, host: str):
+        """Extract host and port from a suite's contact file.
+
+        NB: if it fails to load the suite contact file, it will exit.
+
+        Args:
+            suite (str): suite name
```

Comment:
If using PEP484 notation the whole `(str)` thing becomes superfluous, luckily napoleon supports this, I've not tested it but there are some code examples.

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
I've had a play with this and unfortunately the support is a little disappointing, Napoleon just copies the function signature it doesn't insert the types into brackets for you!!!

I may take a look at Napoleon sometime as that is quite annoying, in the mean time I guess if we do as you have here and document the type twice?

Reply:
>Napoleon just copies the function signature it doesn't insert the types into brackets for you!!!

Disappointing indeed!

>I may take a look at Napoleon sometime as that is quite annoying, in the mean time I guess if we do as you have here and document the type twice?

That'd be fine for me. There are some types that may be complex to represent (especially tuples or mixed lists), where it might be easier to go simply with `tuple`, `list`, or even `object` (pretty much same as not using any PEP-484 `: type`).

Reply:
> But good to know it's still not fully supported!

God Sphinx can be so disappointing some times. I might poke them and see what the plan is but for now...